app/services/data_requests/fetch_cloudwatch_logs.rb

```diff
@@ -39,53 +42,25 @@ def query_string
       QUERY
     end
 
-    def start_queries
-      results = Concurrent::Array.new
-      errors = Concurrent::Array.new
-      thread_pool = Concurrent::FixedThreadPool.new(1, fallback_policy: :abort)
-
-      dates.each do |date|
-        thread_pool.post do
-          warn "Downloading logs for #{date}"
-          results.push(wait_for_query_result(start_query(date)))
-        rescue Aws::CloudWatchLogs::Errors::InvalidParameterException => e
-          if !e.message.match?(/End time should not be before the service was generally available/)
-            errors.push(e)
-          end
-        rescue => e
-          errors.push(e)
-        end
+    def query_ranges
+      dates.map do |date|
+        in_utc = date.in_time_zone('UTC')
+        in_utc.beginning_of_day..in_utc.end_of_day
       end
-
-      thread_pool.shutdown
-      thread_pool.wait_for_termination
-
-      if errors.any?
-        warn "#{errors.count} errors"
-        raise errors.first
-      end
-
-      results
     end
 
-    def start_query(date)
-      query_start = (date - 12.hours).to_i
-      query_end = (date + 36.hours).to_i
```

Comment:
the new code goes from UTC midnight-midnight on each day, it does not have this kind of overlapping padding. Is it worth keeping the padding?

Reply:
yeah I'll see what I can do to join consecutive days into the same query, then add +/- 12 hours to each group.

My worry is a degenerate case where there's an even every day for like 2 months, which risks a query going longer than 15 minutes, so maybe I'll cap each batch at 7 days

Reply:
Yeah, I think that would be great. Even a few days would be a significant improvement I think, not sure where the best maximum is though.

Reply:
updated to do this batching in 316be56bcf2dc459c32e8e4a08fd88f156235401